File: client/java/src/main/java/com/supremainc/sdk/example/card/CardSvc.java
package com.supremainc.sdk.example.card;

import java.util.List;

import com.supremainc.sdk.card.AddBlacklistRequest;
import com.supremainc.sdk.card.DeleteBlacklistRequest;
import com.supremainc.sdk.card.BlacklistItem;
import com.supremainc.sdk.card.CardData;
import com.supremainc.sdk.card.CardGrpc;
import com.supremainc.sdk.card.GetBlacklistRequest;
import com.supremainc.sdk.card.GetBlacklistResponse;
import com.supremainc.sdk.card.ScanRequest;
import com.supremainc.sdk.card.ScanResponse;

public class CardSvc {
  private final CardGrpc.CardBlockingStub cardStub;

  public CardSvc(CardGrpc.CardBlockingStub stub) {
    cardStub = stub;
  }

  public CardData scan(int deviceID) throws Exception {
    ScanRequest request = ScanRequest.newBuilder().setDeviceID(deviceID).build();
    ScanResponse response = cardStub.scan(request);

    return response.getCardData();
  } 

  public List<BlacklistItem> getBlacklist(int deviceID) throws Exception {
    GetBlacklistRequest request = GetBlacklistRequest.newBuilder().setDeviceID(deviceID).build();
    GetBlacklistResponse response = cardStub.getBlacklist(request);

    return response.getBlacklistList();
  }

  public void addBlacklist(int deviceID, List<BlacklistItem> cardInfos) throws Exception {
    AddBlacklistRequest request = AddBlacklistRequest.newBuilder().setDeviceID(deviceID).addAllCardInfos(cardInfos).build();
    cardStub.addBlacklist(request);
  }

  public void deleteBlacklist(int deviceID, List<BlacklistItem> cardInfos) throws Exception {
    DeleteBlacklistRequest request = DeleteBlacklistRequest.newBuilder().setDeviceID(deviceID).addAllCardInfos(cardInfos).build();
    cardStub.deleteBlacklist(request);
  }

}

File: client/csharp/example/card/CardSvc.cs
using System;
using Card;
using Grpc.Core;
using Google.Protobuf.Collections;

namespace example
{
  class CardSvc
  {
    private Card.Card.CardClient cardClient;

    public CardSvc(Channel channel) {
      cardClient = new Card.Card.CardClient(channel);
    }

    public CardData Scan(uint deviceID) {
      var request = new ScanRequest{ DeviceID = deviceID };
      var response = cardClient.Scan(request);

      return response.CardData;
    }

    public RepeatedField<Card.BlacklistItem> GetBlacklist(uint deviceID) {
      var request = new GetBlacklistRequest{ DeviceID = deviceID };
      var response = cardClient.GetBlacklist(request);

      return response.Blacklist;
    }

    public void AddBlacklist(uint deviceID, Card.BlacklistItem[] cardInfos) {
      var request = new AddBlacklistRequest{ DeviceID = deviceID };
      request.CardInfos.AddRange(cardInfos);

      cardClient.AddBlacklist(request);
    }

    public void DeleteBlacklist(uint deviceID, Card.BlacklistItem[] cardInfos) {
      var request = new DeleteBlacklistRequest{ DeviceID = deviceID };
      request.CardInfos.AddRange(cardInfos);

      cardClient.DeleteBlacklist(request);
    }
  }
}